src/target_info.rs

```diff
@@ -64,15 +61,17 @@ fn extract_active_ram_region(
         .cloned()
 }
 
-fn extract_stack_info(
-    elf: &object::read::File,
-    active_ram_region: Option<&RamRegion>,
-    initial_stack_pointer: u32,
-) -> Option<StackInfo> {
-    let active_ram_region = active_ram_region?;
+fn extract_stack_info(elf: &Elf, ram_region: Option<&RamRegion>) -> Option<StackInfo> {
+    // How does it work?
+    // - the upper end of the stack is the initial SP, minus one
+    // - the lower end of the stack is the highest address any section in the elf file uses, plus one
+
+    let ram_range = &ram_region?.range;
```

Comment:
If you want to do some more cleanup: `ram_region?` could be moved out of the function, there isn't really a reason to accept an `Option` here, but then immediately early-return when it's `None`.